src/cache_cf.cc

```diff
@@ -4777,6 +4782,83 @@ static void free_note(Notes *notes)
     notes->clean();
 }
 
+static DebugMessageId ParseDebugMessageId(const char *value, const char eov, const DebugMessageId oldId)
+{
+    if (oldId > 0)
+        throw TexcHere(ToSBuf("repeated id=... or ids=... option in cache_log_message"));
+    const auto id = xatoui(value, eov);
+    if (!(0 < id && id < DebugMessageCount))
+        throw TexcHere(ToSBuf("unknown cache_log_message ID: ", value));
+    return static_cast<DebugMessageId>(id);
+}
+
+static void parse_cache_log_message(DebugMessages *messages)
+{
+    assert(messages);
+
+    DebugMessage msg;
+    DebugMessageId minId = 0;
+    DebugMessageId maxId = 0;
+
+    char *key, *value;
+    while (ConfigParser::NextKvPair(key, value)) {
+        if (strcmp(key, "id") == 0) {
+            minId = maxId = ParseDebugMessageId(value, '\0', minId);
+        } else if (strcmp(key, "ids") == 0) {
+            const auto dash = strchr(value, '-');
+            if (!dash)
+                throw TexcHere(ToSBuf("malformed cache_log_message ID range: ", key, '=', value));
+            const auto oldId = minId;
+            minId = ParseDebugMessageId(value, '-', oldId);
+            maxId = ParseDebugMessageId(dash+1, '\0', oldId);
+            if (minId > maxId)
+                throw TexcHere(ToSBuf("invalid cache_log_message ID range: ", key, '=', value));
+        } else if (strcmp(key, "level") == 0) {
+            const auto level = xatoi(value);
+            if (level < 0)
+                throw TexcHere(ToSBuf("negative cache_log_message level: ", value));
+            msg.level = level;
+        } else if (strcmp(key, "limit") == 0) {
+            msg.limit = xatoull(value, 10);
+        } else {
+            throw TexcHere(ToSBuf("unsupported cache_log_message option: ", key));
+        }
+    }
+
+    if (!minId)
+        throw TexcHere("cache_log_message is missing a required id=... or ids=... option");
+
```

Comment:
I think we should at least warn if `msg` was not touched/configured at all. In fact, I would probably _require_ at least one option (and adjust cf.data.pre BNF accordingly).